src/Microsoft.DotNet.Interactive.Jupyter/Http/IAuthorizationProvider.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Interactive.Jupyter.Http;
+
+internal interface IAuthorizationProvider
+{
+    string GetToken();
+
+    Task<string> GetTokenAsync();
```

Comment:
Why are both sync and async APIs needed?

Reply:
I was thinking interface should support both if we the token needs to be refreshed in async API calls (for e.g. in AAD refresh and SendRequestAsync)